src/components/PopupWithForm.js

```diff
@@ -2,32 +2,44 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super({popupSelector});
+    super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popupElement.querySelector('.modal__form');
+    this._form = this._popupElement.querySelector(".modal__form");
   }
 
   _getInputValues() {
     const inputValues = {};
     this._inputList = this._form.querySelectorAll(".modal__input");
-    this._inputList.forEach(input => {
-      inputValues[input.name] = input.value;  //(???)
+    this._inputList.forEach((input) => {
+      inputValues[input.name] = input.value; //(???)
     });
     return inputValues;
   }
 
-  setEventListeners(){
+  setEventListeners() {
     super.setEventListeners();
-    this._form.addEventListener('submit', (evt) => {
+    this._form.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
       this.close();
     });
   }
 
-close() {
+  close() {
     super.close();
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`